framework/source/class/qx/ui/form/VirtualSelectBox.js

```diff
@@ -593,26 +623,49 @@ qx.Class.define("qx.ui.form.VirtualSelectBox",
 
     __initIncrementalSearch : function()
     {
-      this.__lastRich = this.getChildControl('atom').getRich();
-      this.getChildControl('atom').setRich(true);
       this.__addFilterInput();
-      this.__highlightStyle = this.__getHighlightStyle();
+      this.__highlightStyle = this.isRich() ? this.__getHighlightStyle() : null;
       var that = this;
       var labelOptions = this.getLabelOptions() || {};
       labelOptions.converter = function(data, model, source, target) {
         var filterValue = that.__filterValue;
         if (filterValue && data && (data.toLowerCase().indexOf(filterValue.toLowerCase() != -1))) {
           data = that._highlightFilterValue(data);
         }
+        if (data === undefined) {
+          data = '';
+        }
         return data;
       };
       this.setLabelOptions(labelOptions);
-        var delegate = {
-          configureItem : function(item) {
-            item.setRich(true);
-          }
-      };
-      this.setDelegate(delegate);
+      this.setDelegate({
+        configureItem : function(item) {
+          item.setRich(true);
```

Comment:
The item is set to rich below, in `_applyDelegate` and `_applyRich` as expected. I suspect this is old code that just didn't get deleted, since it sets the item's `rich` property to true regardless of whether the widtget's `rich` property is true.

Reply:
You are right, this must be conditional. It has to be set here in case `setRich()` is called before `setIncrementalSearch()`.
Fixed.